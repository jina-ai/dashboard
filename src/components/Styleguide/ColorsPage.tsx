/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx, useTheme } from "@emotion/react"
import { PageTitle } from "../Common/PageTitle"

const ColorsPage = () => {
  const { palette } = useTheme()
  return (
    <div>
      <PageTitle title={"Colors"} className={""} />
      {Object.keys(palette).map((color) => (
        <div
          key={color}
          css={css`
            padding: 32px;
            background-color: ${palette[color as keyof typeof palette]};
          `}
        >
          <span
            css={css`
              background-color: ${palette.tagBackground};
              color: ${palette.headerTextColor};
              padding: 8px;
              border-radius: 8px;
            `}
          >
            {color} {palette[color as keyof typeof palette]}
          </span>
        </div>
      ))}
    </div>
  )
}

export default ColorsPage
