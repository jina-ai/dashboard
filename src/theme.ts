const PERSIAN_GREEN = "#009999"
const GOLDEN_TANOI = "#FFCC66"
const BITTER_SWEET = "#ff6666"
const SCOOTER_GREEN = "#32C8CD"
const WHITE = "#FFFFFF"
const DADGREY = "#DAD7FE"
const SKY_DIVING_IN_THE_BLUE = "#007AFF"
const QUICK_SILVER = "#A6A6A6"
const HONEYBEE = "#FFCC66"
const BLACK = "#000000"

const theme = {
  palette: {
    background: WHITE,
    headerTextColor: BLACK,
    primary: PERSIAN_GREEN,
    secondary: GOLDEN_TANOI,
    warning: GOLDEN_TANOI,
    error: BITTER_SWEET,
    success: SCOOTER_GREEN,
    info: PERSIAN_GREEN,
    critical: "#ff4540",
    debug: "#7E7278",
    tagBackground: DADGREY,
    tagPlatformBackground: HONEYBEE,
    mutedText: QUICK_SILVER,
    highlight: SKY_DIVING_IN_THE_BLUE,
  },
}

export { theme }
