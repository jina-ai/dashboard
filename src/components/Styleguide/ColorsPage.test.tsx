import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "@emotion/react"
import { theme } from "../../theme"
import ColorsPage from "./ColorsPage"

test("Colors page", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <ColorsPage />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
