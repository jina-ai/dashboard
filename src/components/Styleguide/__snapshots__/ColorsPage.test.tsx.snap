// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors page 1`] = `
<div>
  <div
    className="text-center text-md-left mb-sm-0 col-sm-3 col-12"
  >
    <h3
      className="page-title"
      data-name="title"
    >
      Colors
    </h3>
    <p
      className="page-subtitle mt-2 mb-0"
    />
  </div>
  <div
    className="css-1dttzrv"
  >
    <span
      className="css-13fxtl5"
    >
      background
       
      #FFFFFF
    </span>
  </div>
  <div
    className="css-1yov7ua"
  >
    <span
      className="css-13fxtl5"
    >
      headerTextColor
       
      #000000
    </span>
  </div>
  <div
    className="css-1enftdj"
  >
    <span
      className="css-13fxtl5"
    >
      primary
       
      #009999
    </span>
  </div>
  <div
    className="css-rf90rn"
  >
    <span
      className="css-13fxtl5"
    >
      secondary
       
      #FFCC66
    </span>
  </div>
  <div
    className="css-rf90rn"
  >
    <span
      className="css-13fxtl5"
    >
      warning
       
      #FFCC66
    </span>
  </div>
  <div
    className="css-dt0eag"
  >
    <span
      className="css-13fxtl5"
    >
      error
       
      #ff6666
    </span>
  </div>
  <div
    className="css-hm5zhw"
  >
    <span
      className="css-13fxtl5"
    >
      success
       
      #32C8CD
    </span>
  </div>
  <div
    className="css-1enftdj"
  >
    <span
      className="css-13fxtl5"
    >
      info
       
      #009999
    </span>
  </div>
  <div
    className="css-q4du1v"
  >
    <span
      className="css-13fxtl5"
    >
      critical
       
      #ff4540
    </span>
  </div>
  <div
    className="css-d501cm"
  >
    <span
      className="css-13fxtl5"
    >
      debug
       
      #7E7278
    </span>
  </div>
  <div
    className="css-19tqen3"
  >
    <span
      className="css-13fxtl5"
    >
      tagBackground
       
      #DAD7FE
    </span>
  </div>
  <div
    className="css-rf90rn"
  >
    <span
      className="css-13fxtl5"
    >
      tagPlatformBackground
       
      #FFCC66
    </span>
  </div>
  <div
    className="css-1fad82b"
  >
    <span
      className="css-13fxtl5"
    >
      mutedText
       
      #A6A6A6
    </span>
  </div>
  <div
    className="css-1e4rlla"
  >
    <span
      className="css-13fxtl5"
    >
      highlight
       
      #007AFF
    </span>
  </div>
</div>
`;
